pulp_container/app/tasks/sync_stages.py

```diff
@@ -73,36 +72,20 @@ def __init__(self, remote, signed_only):
         self.deferred_download = self.remote.policy != Remote.IMMEDIATE
         self.signed_only = signed_only
 
+        self.tag_dcs = []
+        self.manifest_list_dcs = []
+        self.manifest_dcs = []
+        self.signature_dcs = []
+
     async def run(self):
         """
         ContainerFirstStage.
         """
 
-        async def get_signature_source(self):
-            """
-            Find out where signatures come from: sigstore, extension API or not available at all.
-            """
-            if self.remote.sigstore:
-                return SIGNATURE_SOURCE.SIGSTORE
-
-            registry_v2_url = urljoin(self.remote.url, "v2")
-            extension_check_downloader = self.remote.get_noauth_downloader(url=registry_v2_url)
-            response_headers = {}
-            try:
-                result = await extension_check_downloader.run()
-                response_headers = result.headers
-            except aiohttp.client_exceptions.ClientResponseError as exc:
-                if exc.status == 401:
-                    response_headers = dict(exc.headers)
-            if response_headers.get(SIGNATURE_HEADER) == "1":
-                return SIGNATURE_SOURCE.API_EXTENSION
-
-        tag_list = []
-        tag_dcs = []
         to_download = []
-        man_dcs = {}
-        signature_dcs = []
-        signature_source = await get_signature_source(self)
+        BATCH_SIZE = 100
```

Comment:
Was the batch size set based on some experience or observation?

Reply:
No. And we should maybe stick to the 500 we use in the stages or do some real measurement.
In the end this batch is there to limit the overall number of high level object (all but Blobs) in the first stage.

One experiment worth doing would be to limit the total number of all items in the pipeline including blobs to limit the amount of memory used.

Reply:
```suggestion
        BATCH_SIZE = 500
```